### PR TITLE
python-sipsimple: 2.6.0 -> 3.0.0 (with all dependencies)

### PR DESCRIPTION
**Pull request by leenaars**
Upgraded 5 Python depencies for Blink Qt (now itself broken) to their current versions. Each dependency has a separate commit. 

Also, because upstream deletes tarballs, moved to directly fetching from the respective upstream darcs repository. One exception: sipsimple cannot yet be fetched from original darcs repo because there is a darcs get flag (--set-scripts-executable) that needs to be set according to [1], and fetchdarcs doesn't seem to be able to do that ...

[1] http://sipsimpleclient.org/projects/sipsimpleclient/wiki/SipInstallation
###### Things done:
- [ ] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- [x] Built on platform(s): NixOS
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @pSub : Blink 2.0.0 has been released, these dependencies are required. As old tarball has been removed upstream (as before), I intend to follow the similar recipe of moving to the upstream darcs repo (now building, will commit separately)

### CODE REVIEW

**Comment by FRidh** [comment]
I've readded the bullet points of the template, please stick to it next time.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -773,13 +773,18 @@ in modules // {
   };
 
   application = buildPythonPackage rec {
-    name = "python-application-${version}";
-    version = "1.5.0";
+    pname = "python-application";
+    name = "${pname}-${version}";
+    version = "2.0.2";
+    disabled = isPy3k;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: why is this disabled? Please add the reason as a comment.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -22213,12 +22222,15 @@ in modules // {
   };
 
   xcaplib = buildPythonPackage rec {
-    name = "python-xcaplib-${version}";
-    version = "1.1.0";
+    pname = "python-xcaplib";
+    name = "${pname}-${version}";
+    version = "1.2.0";
+    disabled = isPy3k;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: same as above

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (leenaars) [inline pkgs/top-level/python-packages.nix]: Py3k is not supported, and although the version currently in nixpkgs is built for it this does not work. 
Running it results in "SyntaxError: Missing parentheses in call to 'print')"

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (leenaars) [inline pkgs/top-level/python-packages.nix]: Py3k is not supported for xcaplib, and although the version currently in nixpkgs is built for it this does not work. 
Running ../bin/xcapclient results in "SyntaxError: Missing parentheses in call to 'print')"

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (leenaars) [inline pkgs/top-level/python-packages.nix]: python-application does not support Py3k; is full of Python 2 specific legacy code.

**Comment by leenaars** [comment]
I've now succeed in building blink 2.0.0 with all dependencies. There is a new dependency on sipsimple called python-otr which I've added in a separate pull request. This dependency has not yet been added to sipsimple simply because I don't know which order to best submit this in...

**Comment by FRidh** [comment]
Great, good job. Typically it is easiest if commits that depend on each other are in the same PR. If one of the commits has a big impact, it is often better to consider that one separate.

I couldn't find any other package that depends on `python-gnutls` with `git grep python-gnutls`. So go ahead, update the packages and add those updates that are needed for this PR to this PR.

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: Okay, reasons like this are best included in the code whenever disabling packages or tests.

**Comment by leenaars** [comment]
This has the whole chain of dependencies (which indeed I don't think are used by other applications) up and including blinkqt, which (for me) now builds.

File: pkgs/applications/networking/instant-messengers/blink/pythonpath.patch [hunk 3]
```diff
@@ -1,14 +1,44 @@
-diff -rupN a/blink/resources.py b/blink/resources.py
---- a/blink/resources.py	2015-03-17 03:24:06.000000000 -0600
-+++ b/blink/resources.py	2015-04-07 22:52:06.101096413 -0600
-@@ -60,14 +60,7 @@ class Resources(object):
+--- blink-2.0.0/blink/resources.py	2016-03-09 14:39:07.000000000 +0100
++++ blink-2.0.0/blink/resources-patched.py	2016-03-12 21:34:14.965476623 +0100
+@@ -1,7 +1,10 @@
++# Copyright (C) 2010-2013 AG Projects. See LICENSE for details.
```
Comment (FRidh) [inline pkgs/applications/networking/instant-messengers/blink/pythonpath.patch]: This got me curious (nothing wrong with it), so I checked LICENSE and noticed that this version of Blink has a different license than is mentioned in the Nix expression. 
http://devel.ag-projects.com/repositories/blink-qt/LICENSE
Apparently it is GPL3 + additional restrictions.
Maybe @jagajaga nows what we do in such cases?

Comment (leenaars) [inline pkgs/applications/networking/instant-messengers/blink/pythonpath.patch]: It is GPLv3+, indeed. I know the developers, and it has always been so. I don't understand why the mit license was there ... probably a copying error...

I thought I could update that tomorrow, together with adding a desktop icon. Going to sleep now ...

**Comment by FRidh** [comment]
ping

**Comment by leenaars** [comment]
Pong @FRidh: I'm wondering what I should do next: I have updated all the dependencies, the desktop icon and license are included in my latest commit. 

I've reached out to the project to ask them to make it clear in their description that they are now dual licensing (GPLv3 without constraints, and BSD-like with some additional constraints). I'm still awaiting their feedback.

**Comment by FRidh** [comment]
Thanks. It would indeed be nice if they would be a bit clearer with the license.

**Pull request merged**